### PR TITLE
[develop] Add custom munge key config logic

### PR DESCRIPTION
**Pull request by hehe7318**
### Description of changes
* Create a munge key manager
* Add related kitchen tests

### Relate Pull Request
* See the PR of Setup custom munge key in cli: [[develop] Cli Setup custom munge key](https://github.com/aws/aws-parallelcluster/pull/5681)
* [[develop] Add custom munge key update logic](https://github.com/aws/aws-parallelcluster-cookbook/pull/2452)
* [[develop] Add custom munge key rotation script](https://github.com/aws/aws-parallelcluster-cookbook/pull/2453)

### New sample YAML configuration

```
DevSettings:
  SlurmSettings:
    MungeKeySecretArn: Str
```

### CODE REVIEW

File: cookbooks/aws-parallelcluster-slurm/kitchen.slurm-config.yml [hunk 1]
```diff
@@ -176,3 +176,19 @@ suites:
...
+      - /tag:config_munge/
+    attributes:
+      node_type: HeadNode
+      scheduler: 'slurm'
+      cluster:
+        region: 'us-east-2'
+        config:
+          DevSettings:
+            SlurmSettings:
+              MungeKeySecretArn: 'arn:aws:secretsmanager:us-east-2:249582277112:secret:TestMungeKey-jDtnqi'
```
Comment (EddyMM) [inline cookbooks/aws-parallelcluster-slurm/kitchen.slurm-config.yml]: This configuration will only work for this specific account and region. Let's sync on how to proceed.

File: cookbooks/aws-parallelcluster-slurm/resources/munge_key_manager.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+default_action :setup_munge_key
+
+action :setup_munge_key do
+  if new_resource.munge_key_secret_arn
+    # This block will fetch the munge key from Secrets Manager
+    bash 'fetch_and_decode_munge_key' do
+      user 'root'
+      group 'root'
+      cwd '/tmp'
+      code <<-FETCH_AND_DECODE
```
Comment (EddyMM) [inline cookbooks/aws-parallelcluster-slurm/resources/munge_key_manager.rb:33]: Add `set -e` to avoid script completion if a step fails

Comment (hehe7318) [reply cookbooks/aws-parallelcluster-slurm/resources/munge_key_manager.rb:33]: Done!

File: cookbooks/aws-parallelcluster-slurm/kitchen.slurm-config.yml (hunk 1 above)
Comment (hehe7318) [reply cookbooks/aws-parallelcluster-slurm/kitchen.slurm-config.yml]: Done!

File: cookbooks/aws-parallelcluster-slurm/kitchen.slurm-config.yml [hunk 3]
```diff
@@ -176,3 +176,19 @@ suites:
       cluster:
         node_type: ComputeFleet
         head_node_private_ip: '127.0.0.1'
+  - name: config_head_node_munge
+    run_list:
+    - recipe[aws-parallelcluster-tests::setup]
+    - recipe[aws-parallelcluster-slurm::config_munge_key]
+    verifier:
+    controls:
```
Comment (EddyMM) [inline cookbooks/aws-parallelcluster-slurm/kitchen.slurm-config.yml]: `controls` should be nested under `verifier`.

**Review by EddyMM — APPROVED** [review]

**Pull request merged**